### PR TITLE
[Structural] Shell q4thk initial strain test

### LINKED ISSUE #9273: Implementing initial strains/stresses in shell elements
In order to simulate thermal expansion/contraction in structures we need to implement initial strains/stresses in shell and beam elements (we see that they are already implemented for solid elements.)
Starting with shell elements, it seems that the implementation should be quite straightforward, because the constitutive behaviour is defined using the `ConstitutiveLaw` class at the level of the integration points of the ply. Therefore a modification of `set_initial_state_process.cpp` for the case of shell elements to point to the ply integration points should suffice to achieve the desired behaviour.
In pseudo-code this should read something like this:
~~~c++
if (r_element.HasCrossSections()){
    loop for the sections in the element
        loop for the plies in the section
            loop for the integration points
                constitutive_law_vector[point_number]->SetInitialState(p_initial_state);

}
else{
    execute-existing-code
}
~~~
We would be glad if we get some support for this implementation, as we are Kratos newbies.
In particular we get stuck at how to point to the sections of the shell element, and how to point to the plies of the section.
For the first case it seems that a public method like
~~~c++
CrossSectionContainerType const& GetCrossSectionsOnIntegrationPoints() const
{
    return *mSections;
}
~~~
should be added to `base_shell_element.hpp` (and possibly to `element.h` as virtual method??)
And something similar for the plies:
~~~c++
PlyCollection const& GetPlyCollection() const
{
    return mStack;
}
~~~
We will be grateful for any help with this enhancement!

### PR DESCRIPTION
**Pull request by clazaro**
**📝 Description**
Adding a test of initial strains in a Q4 thick shell element.

**🆕 Changelog**
Added test files into StructuralMechanicsApplication/tests/InitialStateElasticity/
Added the corresponding class in structural_mechanics_test_factory.py

I didn't add the test into test_StructuralMechanicsApplication.py as I wasn't sure if it has to be included into the smallSuite or the nightSuite.

### CODE REVIEW

**Comment by clazaro** [comment]
Apologies for the previous failed (closed) PRs...
Now I got it!

File: applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py [hunk 1]
```diff
@@ -239,6 +239,9 @@ class Simple2D2NBeamCrTest(StructuralMechanicsTestFactory):
 class InitialStateElasticityTest(StructuralMechanicsTestFactory):
     file_name = "InitialStateElasticity/initial_state_test"
 
+class InitialStrainShellQ4ThickTest(StructuralMechanicsTestFactory):
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py:242]: I think the test is still missing in test_StructuralMechanicsApplication.py (maybe is added automatically, last I checked it wasn't)

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Thanks, add to test_StructuralMechanicsApplication.py, otherwise will not be computed in CI

**Comment by clazaro** [comment]
@loumalouomega, added

**Review by loumalouomega — APPROVED** [review]
Okidoki

File: applications/StructuralMechanicsApplication/tests/InitialStateElasticity/initial_strain_shell_Q4_thick_test_parameters.json [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            "mesh_id"         : 0,
+            "model_part_name" : "Structure.Parts_Shell_Slab",
+            "dimension"       : 2,
+            "imposed_strain"  : [1.0e-4,-5.0e-5,-2e-4],
+            "imposed_stress"  : [0,0,0],
+            "imposed_deformation_gradient" : [[1,0,0],[0,1,0],[0,0,1]],
+            "interval"        : [0.0,"End"]
+            }
+        }],
+        "_print_output_process" : [{
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/InitialStateElasticity/initial_strain_shell_Q4_thick_test_parameters.json]: the underscore syntax is not reliable (in terms of it works in some cases and not in others) and is from a time before we could have comments in the json

Hence I would comment this block. Commenting works same as in C++

What happens here is that this process still gets executed and hence renders the test useless as it constantly writes new reference results

File: applications/StructuralMechanicsApplication/tests/InitialStateElasticity/initial_strain_shell_Q4_thick_test_parameters.json [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                "model_part_name"   : "Structure.Parts_Shell_Slab",
+                "check_variables"   : ["DISPLACEMENT"],
+                "input_file_name"   : "InitialStateElasticity/initial_strain_shell_Q4_thick_test_results.json",
+                "time_frequency"    : 1.0
+            }
+        }
+        ],
+        "list_other_processes"     : []
+    },
+    "_output_processes" : {
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/InitialStateElasticity/initial_strain_shell_Q4_thick_test_parameters.json]: same

**Review by philbucher — COMMENTED** [review]
minor but important comment

**Review by philbucher — APPROVED** [review]

**Comment by clazaro** [comment]
Done

**Pull request merged**